File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/SoundEffectPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.network.packet.Packet;

import io.netty.buffer.ByteBuf;

public final record SoundEffectPacket(String soundName, int x, int y, int z, float volume, short pitch)
        implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        writeString(os, soundName);
        os.writeInt(x);
        os.writeInt(y);
        os.writeInt(z);
        os.writeFloat(volume);
        os.writeByte(pitch);
    }

    @Override
    public byte getId() {
        return 0x29;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/SoundEffectPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class SoundEffectPacket(
    val soundName: String,
    val x: Int,
    val y: Int,
    val z: Int,
    val volume: Float,
    val pitch: Short
) : Packet.Outgoing {
    override fun serialize(os: ByteBuf) {
        os.writeString(soundName)
        os.writeInt(x, y, z)
        os.writeFloat(volume)
        os.writeByte(pitch.toInt())
    }

    override val id: Byte
        get() = 0x29
}
